Add wgscoverageplotter from jvarkit

pvanheus opened the pull request:
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

mvdbeek commented:
Can you add a `.shed.yaml` file @pvanheus ?

wm75 commented:
@pvanheus do you really need picard here just to generate the sequence dict or would something like `samtools view -H` | `grep` work, too?

pvanheus commented:
> @pvanheus do you really need picard here just to generate the sequence dict or would something like `samtools view -H` | `grep` work, too?
You mean something like `samtools -H '$alignment_file' | grep -E '^(@HD|@SQ)' >reference.dict` ? I.e. extract the dictionary from the BAM header, not the FASTA? Does that do the same as `picard` ?

wm75 commented on tools/jvarkit_wgscoverageplotter/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,236 @@
+        --output plot.svg
+        #if str($adv.adv_choice) == "true"
+            $adv.disable_paired_overlap
+            #if len($include_regex) != 0
```
```suggestion
            #if $include_regex
```

wm75 commented on tools/jvarkit_wgscoverageplotter/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,236 @@
+            #if len($include_regex) != 0
+                -I '$include_regex'
+            #end if
+            #if len($exclude_regex) != 0
```
```suggestion
            #if $exclude_regex
```

wm75 commented on tools/jvarkit_wgscoverageplotter/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,236 @@
+            --min-contig-length $adv.min_contig_length
+            --percentile $adv.percentile
+            $adv.points
+            #if len(str($adv.samples).strip()) != 0
```
```suggestion
            #if str($adv.samples).strip()
```

wm75 replied:
Actually, this one will turn into a validator on the param if you pick up my suggestion of a conditional block around --samples and --partition.

wm75 commented on tools/jvarkit_wgscoverageplotter/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,236 @@
+            <option value="SVG">SVG</option>
+        </param>
+        <conditional name="adv">
+            <param type="select" name="adv_choice" label="Show advanced options">
```
I guess a `<section>` is the modern style here instead of a `conditional`.

wm75 commented on tools/jvarkit_wgscoverageplotter/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,236 @@
+                    type="text"
+                    label="Limit to these read groups" 
+                    help="See also --partition. Multiple read groups can be separated with commas." />
+                <param argument="--partition" type="select" label="Partition data using SAM Read Groups (requires --samples)">
```
Since this option requires `--samples` I would move things into a conditional that asks whether to limit to certain read groups -> only <when "yes"> -> bring up --samples and --partition

pvanheus replied:
I have been trying to figure out how `--samples` and `--partition` work, but failed, so I have opened an [issue](https://github.com/lindenb/jvarkit/issues/177) on the upstream project and have disabled these options in the wrapper for now.

wm75 replied:
Seems like a good decision!

pvanheus replied:
I got a speedy reply from Pierre Lindenbaum clarifying the use of `--samples` and `--partition` and have added these options back in the wrapper as per your suggestion @wm75.

wm75 reviewed (commented):
Looks pretty good already. Just some minor improvement suggestions.

bernt-matthias commented:
What do you think about renaming the folder to jvarkit.. Just in case someone wants to add more tools from this suite someday? Maybe also use a suite style shed.yml..

pvanheus commented:
> What do you think about renaming the folder to jvarkit.. Just in case someone wants to add more tools from this suite someday? Maybe also use a suite style shed.yml..

Good idea, I've switched to a suite style `shed.yml` and renamed the directory to `jvarkit`.

wm75 commented on tools/jvarkit/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,227 @@
+        --min-contig-length $adv.min_contig_length
+        --percentile $adv.percentile
+        $adv.points
+        #if str($adv.sample_filter.filter_by_sample) == "true" and str($adv.sample_filter.samples).strip()
```
I wouldn't do the check for an empty --samples param here, but rather have a `<validator>` on the param itself.

If a user wants filtering by sample, but doesn't enter a sample identifier, then either that's a typo or they didn't understand what they are doing. In either case this shouldn't be passed over silently, but should be caught and brought to the user's attention as early as possible.

wm75 commented on tools/jvarkit/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,227 @@
+                    <option value="false" selected="true">No</option>
+                </param>
+                <when value="true">
+                    <param argument="--samples"
```
This should be validated and not allowed to be empty after stripping whitespace.

wm75 commented on tools/jvarkit/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,227 @@
+                type="integer" 
+                label="Skip chromosomes shorter than"
+                value="0"
+                help="A distance specified as a positive integer. Commas are removed. The following suffixes are interpreted: b,bp,k,kb,m,mb" />
```
```suggestion
                min="0"
                help="A distance specified as a positive integer." />
```

More is not possible with an integer param.

wm75 commented on tools/jvarkit/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,227 @@
+                        type="text"
+                        label="Limit to these read groups" 
+                        help="See also --partition. Multiple read groups can be separated with commas." />
+                    <param argument="--partition" type="select" label="How to interpret identifiers in --samples" help="Partitions are modelled on the GATK login as described here: https://gatk.broadinstitute.org/hc/en-us/articles/360051307491-DepthOfCoverage-BETA-#--partition-type">
```
```suggestion
                    <param argument="--partition" type="select" label="How to interpret identifiers in --samples" help="Partitions are modelled on the GATK logic as described here: https://gatk.broadinstitute.org/hc/en-us/articles/360051307491-DepthOfCoverage-BETA-#--partition-type">
```

wm75 commented on tools/jvarkit/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,227 @@
+            <param argument="--max-depth" 
+                name="max_depth" 
+                type="integer" 
+                label="Max depth to display." 
```
I know that's the help of the command line tool, but maybe 'y-axis (depth) limit of the plot' or similar would be clearer?

wm75 commented on tools/jvarkit/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,227 @@
+                help="The special value '-1' will first compute the average depth and the set the max depth to 2*average" />
+            <param argument="--clip"
+                type="boolean"
+                label="Don't show coverage to be greater than 'max-depth' in the output image"
```
Similar to the above: maybe sth like label="Clip the plot at the y-axis limit" help="Default is to let the coverage plot exceed the y-axis limit"?

This would also avoid the double-negation in the default situation, i.e.:
*Don't show coverage to be greater than 'max-depth' in the output image*: `No`
which always makes my head spin :)

bgruening commented on tools/jvarkit/.shed.yml:
```diff
@@ -0,0 +1,17 @@
+  name_template: "{{ tool_id }}"
+  description_template: "Wrapper for {{ tool_name }}"
+suite:
+  name: "suite_varscan"
```
```suggestion
  name: "suite_jvarkit"
```

bgruening commented on tools/jvarkit/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,232 @@
+        #if str($format) == "SVG"
+            mv plot.svg '$plot_output'
+        #else
+            convert plot.svg '$format:$plot_output'
```
@pvanheus do we need this complexity inside this tool? We have an ImageMagick tool that can do this in a separate step. This is not a -1 just a question.

pvanheus replied:
If we leave the PNG support out here then, for Galaxy servers without SVG display support (I think a lot of them at this point?) it means always having to run 2 tools to get an image. So the cost of simplifying this tool comes with a cost of making its use more complex, and thus I'd prefer to leave this in.

bgruening commented on tools/jvarkit/jvarkit_wgscoverageplotter.xml:
```diff
@@ -0,0 +1,232 @@
+        samtools view -H '$alignment_file' |grep -E '^(@HD|@SQ)' >reference.dict &&
+        cat reference.dict &&
+        ln -s '$alignment_file' '$alignment_name' &&
+        ln -s '${alignment_file.metadata.bam_index}' '$alignment_name'.bai &&
```
```suggestion
        ln -s '${alignment_file.metadata.bam_index}' '$alignment_name.bai' &&
```

wm75 commented:
Thanks a lot @pvanheus for this very useful wrapper!
